sdk/servicebus/service-bus/src/core/linkEntity.ts

```diff
@@ -268,4 +310,109 @@ export class LinkEntity {
     }
     return result;
   }
+
+  /**
+   * Determines whether the AMQP link is open. If open then returns true else returns false.
+   * @return {boolean} boolean
+   */
+  isOpen(): boolean {
+    const result: boolean = this._link ? this._link.isOpen() : false;
+    log.error(`${this._logPrefix} is open? ${result}`);
+    return result;
+  }
+
+  /**
+   * Indicates that _closeLink("close") has been called on this link and
+   * that it should not be allowed to reopen.
+   */
+  private _wasClosedByUser: boolean = false;
+
+  protected get wasClosedByUser(): boolean {
+    return this._wasClosedByUser;
+  }
+
+  /**
+   * Indicates that a link initialization is in process.
+   */
+  public get isConnecting(): boolean {
+    return this._isConnecting;
+  }
+
+  private _isConnecting: boolean = false;
```

Comment:
Any reason to have these private variables and public getters in this part of the file vs before the constructor which seems to be the pattern used in other places in the package?

Reply:
Will fix.

Reply:
Sorted-ish now.

public fields, protected fields, private fields
constructor
public methods/properties, protected, private